### ISSUE TITLE
Page titles are not showing

### ISSUE THREAD

**Issue by carrieforde** [carrieforde/Alcatraz#74]
Page titles do not show on pages (regardless of hide title being unchecked), and leave empty `<header>` tags in the markup.

**Comment by BraadMartin** [comment]
I commented on the commit, but it didn't get included here, so here is a paste:

Hmm, the original logic here was supposed to be "For all singular pages, check $hide_title and either output nothing if it is set to true or output an `<h1>`, then for all other pages output an `<h2>`"

The old logic was actually saying "If we're on a search results page and not a singular page, output an `<h2>`, otherwise check whether $hide_title is set and maybe output an `<h1>`". So it was a bit backwards, but the goal was that the `is_search() || ! is_singular()` check would tell us whether we're not on a singular page (mostly this logic was left over from Underscores, which checks is_search explicity).

The new logic says "If we're on a search results or singular page, use an `<h1>`, but then if $hide_title is set, regardless of which context we're in either output nothing or an `<h1>` if it's not singular". The problem with this logic is that it switches away from the elseif, and technically the `<h1>` can never get output, because it will only even get checked for if $hide_title is set to true, which would cause no title to get output.

Let's try

```
if ( is_singular() ) {
    // Check $hide_title and output an <h1> if it is false or not set
} else {
    // Output <h2>
}
```

This is much simpler and is a more direct mapping to what we want, I think. The point is we only want to output the `<h1>` on singular pages, and we only want maybe hide the title on singular pages. For all other use cases we want the `<h2>`.

**Comment by carrieforde** [comment]
Fixed by @BraadMartin with d240d96c443e25c886583c6b2523bb832e52e57c
